plonky2/src/fri/oracle.rs

```diff
@@ -154,6 +154,10 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
             alpha.shift_poly(&mut final_poly);
             final_poly += quotient;
         }
+        final_poly.trim();
```

Comment:
Seems like it's possible this would result in a non-power-of-two size, if there were multiple `0`s at the end. How about using `rotate_right(1)` (and maybe asserting that a `0` was rotated)? To ensure the size doesn't change

Reply:
Hmm good point.
Now that I look at it, the 
```rust
quotient.padded(quotient.degree_plus_one().next_power_of_two())
```
in `compute_quotient` seems a bit weird, especially now that we only divide by linear factors. Maybe we could remove this padding, remove the `final_poly.trim()` here, and keep the `final_poly.coeffs.insert(0,0)` to multiply by `X`.
What do you think?

Reply:
Sounds good to me 👍